### PR TITLE
test: Fix server tests

### PR DESCRIPTION
**Pull request by colinkiama**
Required for: https://github.com/valum-framework/valum/issues/226

### CODE REVIEW

File: tests/meson.build [hunk 1]
```diff
@@ -14,7 +14,8 @@ vsgi_tests = [
 foreach name : vsgi_tests
     test(name, executable(name + '-test', name + '-test.vala',
                           dependencies: [glib, gobject, gio, gio_unix, soup, vsgi_dep]),
-         env: ['G_TEST_SRCDIR=' + meson.current_source_dir()])
+         env: ['G_TEST_SRCDIR=' + meson.current_source_dir(),
+               'VSGI_SERVER_PATH=' + join_paths(meson.current_build_dir(), '../src/servers')])
```
[the remark this answers is not in the document]
Comment (colinkiama) [reply tests/meson.build:18]: After debugging library searches with `LD_DEBUG=libs`, It seems like `build_rpath` only applies for finding libraries before an application starts.

When attempting to load modules with `GLib.Module.open ()`, only system paths are searched. Nothing from build_rpath is included. 

Here's a log of what happens when when attempting to load the http-server module in the http-server test :
```
# Start of server tests
# rpath: (null)
     49034:	find library=libvsgi-http.so [0]; searching
     49034:	 search cache=/etc/ld.so.cache
     49034:	 search path=/lib/x86_64-linux-gnu/tls/haswell/x86_64:/lib/x86_64-linux-gnu/tls/haswell:/lib/x86_64-linux-gnu/tls/x86_64:/lib/x86_64-linux-gnu/tls:/lib/x86_64-linux-gnu/haswell/x86_64:/lib/x86_64-linux-gnu/haswell:/lib/x86_64-linux-gnu/x86_64:/lib/x86_64-linux-gnu:/usr/lib/x86_64-linux-gnu/tls/haswell/x86_64:/usr/lib/x86_64-linux-gnu/tls/haswell:/usr/lib/x86_64-linux-gnu/tls/x86_64:/usr/lib/x86_64-linux-gnu/tls:/usr/lib/x86_64-linux-gnu/haswell/x86_64:/usr/lib/x86_64-linux-gnu/haswell:/usr/lib/x86_64-linux-gnu/x86_64:/usr/lib/x86_64-linux-gnu:/lib/tls/haswell/x86_64:/lib/tls/haswell:/lib/tls/x86_64:/lib/tls:/lib/haswell/x86_64:/lib/haswell:/lib/x86_64:/lib:/usr/lib/tls/haswell/x86_64:/usr/lib/tls/haswell:/usr/lib/tls/x86_64:/usr/lib/tls:/usr/lib/haswell/x86_64:/usr/lib/haswell:/usr/lib/x86_64:/usr/lib		(system search path)
     49034:	  trying file=/lib/x86_64-linux-gnu/tls/haswell/x86_64/libvsgi-http.so
     49034:	  trying file=/lib/x86_64-linux-gnu/tls/haswell/libvsgi-http.so
     49034:	  trying file=/lib/x86_64-linux-gnu/tls/x86_64/libvsgi-http.so
     49034:	  trying file=/lib/x86_64-linux-gnu/tls/libvsgi-http.so
     49034:	  trying file=/lib/x86_64-linux-gnu/haswell/x86_64/libvsgi-http.so
     49034:	  trying file=/lib/x86_64-linux-gnu/haswell/libvsgi-http.so
     49034:	  trying file=/lib/x86_64-linux-gnu/x86_64/libvsgi-http.so
     49034:	  trying file=/lib/x86_64-linux-gnu/libvsgi-http.so
     49034:	  trying file=/usr/lib/x86_64-linux-gnu/tls/haswell/x86_64/libvsgi-http.so
     49034:	  trying file=/usr/lib/x86_64-linux-gnu/tls/haswell/libvsgi-http.so
     49034:	  trying file=/usr/lib/x86_64-linux-gnu/tls/x86_64/libvsgi-http.so
     49034:	  trying file=/usr/lib/x86_64-linux-gnu/tls/libvsgi-http.so
     49034:	  trying file=/usr/lib/x86_64-linux-gnu/haswell/x86_64/libvsgi-http.so
     49034:	  trying file=/usr/lib/x86_64-linux-gnu/haswell/libvsgi-http.so
     49034:	  trying file=/usr/lib/x86_64-linux-gnu/x86_64/libvsgi-http.so
     49034:	  trying file=/usr/lib/x86_64-linux-gnu/libvsgi-http.so
     49034:	  trying file=/lib/tls/haswell/x86_64/libvsgi-http.so
     49034:	  trying file=/lib/tls/haswell/libvsgi-http.so
     49034:	  trying file=/lib/tls/x86_64/libvsgi-http.so
     49034:	  trying file=/lib/tls/libvsgi-http.so
     49034:	  trying file=/lib/haswell/x86_64/libvsgi-http.so
     49034:	  trying file=/lib/haswell/libvsgi-http.so
     49034:	  trying file=/lib/x86_64/libvsgi-http.so
     49034:	  trying file=/lib/libvsgi-http.so
     49034:	  trying file=/usr/lib/tls/haswell/x86_64/libvsgi-http.so
     49034:	  trying file=/usr/lib/tls/haswell/libvsgi-http.so
     49034:	  trying file=/usr/lib/tls/x86_64/libvsgi-http.so
     49034:	  trying file=/usr/lib/tls/libvsgi-http.so
     49034:	  trying file=/usr/lib/haswell/x86_64/libvsgi-http.so
     49034:	  trying file=/usr/lib/haswell/libvsgi-http.so
     49034:	  trying file=/usr/lib/x86_64/libvsgi-http.so
     49034:	  trying file=/usr/lib/libvsgi-http.so
     49034:	
Bail out! FATAL-CRITICAL: vsgi-server-module.vala:68: libvsgi-http.so: cannot open shared object file: No such file or directory

** (./server-test:49034): CRITICAL **: 13:22:57.352: vsgi-server-module.vala:68: libvsgi-http.so: cannot open shared object file: No such file or directory
Trace/breakpoint trap

```

**Comment by colinkiama** [comment]
All test pass now without any modifications made to the test executables.

Here's more info about the default `-rpath` behaviour changes: https://stackoverflow.com/questions/52018092/how-to-set-rpath-and-runpath-with-gcc-ld

**Comment by arteymix** [comment]
So yeah, apparently `-rpath` now means `-runpath`. It's not transitive anymore, that's why the test binary is broken. Not that it is recommended though to use `-rpath` because it cannot be overwritten by the end-user.

We can address this differently though via pkg-config (maybe `Lib.private`? I need to review this) and redefining `vsgi_dep` to add the servers' folder to the dynamic linker. I'll merge this however since it address the regression.

Also your text editor is removing the last newline character of the file. We should copy over Valum's EditorConfig setup in this repository.

File: src/meson.build [hunk 2]
```diff
@@ -50,4 +51,4 @@ pkgconfig.generate(vsgi_lib,
                    description: 'Interface and implementations for various web server technologies',
                    subdirs: 'vsgi-' + api_version)
 
-subdir('servers')
+subdir('servers')
```
Comment (arteymix) [inline src/meson.build:54]: Please fix your text editor!

**Review by arteymix — CHANGES_REQUESTED** [review]

**Comment by arteymix** [comment]
Also, you can create branches in this repo for PRs since you have the privileges for that. It's a bit more convenient for me if I want to checkout your code.

**Pull request merged**